<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Performs Bayesian guided sparse factor analysis on gene expression 
   matrix with matching sample-level genetic perturbation covariates.">
<title>Bayesian Guided Sparse Factor Analysis • GSFA</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Bayesian Guided Sparse Factor Analysis">
<meta property="og:description" content="Performs Bayesian guided sparse factor analysis on gene expression 
   matrix with matching sample-level genetic perturbation covariates.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">GSFA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/gsfa_intro.html">Applying GSFA in a simulated dataset</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/xinhe-lab/GSFA/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gsfa">GSFA<a class="anchor" aria-label="anchor" href="#gsfa"></a>
</h1></div>
<p><code>GSFA</code> (Guided Sparse Factor Analysis) is an R package (accelerated by Rcpp) that performs sparse factor analysis and differential gene expression discovery simultaneously on single-cell RNA-seq with multiplexed CRISPR screen (e.g. CROP-seq, Perturb-seq) data.</p>
<p>The method assumes that the perturbation of a target gene affects certain latent factors (gene modules), which in turn alters the expression of individual genes. By fitting a Bayesian hierarchical model using Gibbs sampling, GSFA infers the latent factors and their associations with the perturbations in a joint statistical framework. It also summarizes the effects of a perturbation on individual genes as the sum of effects mediated by all the factors.</p>
<p>Provided with a normalized gene expression matrix and a perturbation matrix (of which cells contain which type of gRNAs) a single-cell CRISPR screening experiment, <code>GSFA</code> can<br>
(1) identify genetically controlled factors that are associated with the perturbation;<br>
(2) interpret the biological meanings of factors (gene modules) through gene ontology/pathway enrichment analysis thanks to the sparse gene weights on factors;<br>
(3) detect differentially expressed genes under each genetic perturbation by thresholding the local false sign rate (<em>LFSR</em>).</p>
<p><img src="reference/figures/schematic.png"></p>
<div class="section level2">
<h2 id="citing-this-work">Citing this work<a class="anchor" aria-label="anchor" href="#citing-this-work"></a>
</h2>
<p>If you find the GSFA package or any of the source code in this repository useful for your work, please cite:</p>
<blockquote>
<p>Yifan Zhou, Kaixuan Luo, Lifan Liang, Mengjie Chen and Xin He. A new Bayesian factor analysis method improves detection of genes and biological processes affected by perturbations in single-cell CRISPR screening. <em>Nature Methods</em>. (2023). doi: 10.1038/s41592-023-02017-4. PMID: 37770710.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>All source code and software in this repository are made available under the terms of the <a href="https://opensource.org/licenses/mit-license.html" class="external-link">MIT license</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="github">Github<a class="anchor" aria-label="anchor" href="#github"></a>
</h3>
<p>To install the development version of the <code>GSFA</code> package from Github, run this in R:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"xinhe-lab/GSFA"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Note that installing the package will require a C++ compiler setup that is appropriate for the version of R installed on your computer.</p>
</div>
<div class="section level3">
<h3 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a>
</h3>
<p>If you have cloned the repository locally, you can install the package with the <code>install_local</code> function from devtools. Assuming you are in the local GSFA repository, run this code in R to install the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_local.html" class="external-link">install_local</a></span><span class="op">(</span>build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Note that installing the package will require a C++ compiler setup that is appropriate for the version of R installed on your computer.</p>
</div>
<div class="section level3">
<h3 id="docker">Docker<a class="anchor" aria-label="anchor" href="#docker"></a>
</h3>
<p>If you are familiar with Docker, this R package and all its C++ and R dependencies built on <code>debian stable</code> have been containerized in a docker image. Run the code below to pull the docker image of the release version of <code>GSFA</code>. For more details on the dependencies, please see <code>Dockerfile</code> in this repository.</p>
<pre><code>docker pull gradonion/gsfa:latest</code></pre>
<p>Once pulled, run the docker image (by modifying the <code>docker run</code> command below) and launch RStudio Server locally at <a href="localhost:8787">localhost:8787</a> to run the vignettes or your own analyses.</p>
<pre><code>docker run \
    -d \
    -e DISABLE_AUTH=true \
    -v /your_local_directory/:/home/rstudio/projects/ \
    -p 8787:8787 \
    gradonion/gsfa:latest</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="using-the-package">Using the package<a class="anchor" aria-label="anchor" href="#using-the-package"></a>
</h2>
<p>Please see this package <a href="https://xinhe-lab.github.io/GSFA/articles/gsfa_intro.html" class="external-link">vignette</a> for using GSFA on a simulated example:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/xinhe-lab/GSFA" class="external-link">GSFA</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"gsfa_intro"</span><span class="op">)</span></span></code></pre></div>
<p>For guidance on using GSFA to analyze and interpret real single-cell CRISPR screen data, please refer to the code and examples in this <a href="https://github.com/xinhe-lab/GSFA_paper" class="external-link">repository</a>.</p>
</div>
<div class="section level2">
<h2 id="credits">Credits<a class="anchor" aria-label="anchor" href="#credits"></a>
</h2>
<p>The GSFA package is developed by Yifan Zhou from the <a href="http://xinhelab.org" class="external-link">He Lab</a> and <a href="http://www.mengjiechen.com" class="external-link">Chen Lab</a> at the University of Chicago.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/xinhe-lab/GSFA/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/xinhe-lab/GSFA/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GSFA</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Yifan Zhou <br><small class="roles"> Maintainer, author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Yifan Zhou.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
